sdk/src/asset_handlers/sfnt_io.rs

```diff
@@ -179,15 +179,11 @@ mod font_xmp_support {
     /// offers a more generic method to indicate a document ID, instance ID,
     /// and a reference to the a remote manifest.
     #[allow(dead_code)]
-    pub(crate) fn add_reference_as_xmp_to_stream<TSource, TDest>(
-        source: &mut TSource,
-        destination: &mut TDest,
+    pub(crate) fn add_reference_as_xmp_to_stream(
+        source: &mut dyn CAIRead,
```

Comment:
Noob question, but why is this `dyn CAIRead` and not `impl CAIRead`?

Reply:
The `impl CAIRead` is basic syntax sugar for generic parameters. And I am not going to claim I fully understand this, but if you use generic parameters in the trait, it cannot "safely" create object. So using the `dyn CAIRead` allows for dispatching execution instead. 

See https://doc.rust-lang.org/reference/items/traits.html#object-safety for more details.